colossalai/booster/plugin/gemini_plugin.py

```diff
@@ -62,6 +73,54 @@ def save_lr_scheduler(self, lr_scheduler: LRScheduler, checkpoint: str):
         if self.coordinator.is_master():
             super().save_lr_scheduler(lr_scheduler, checkpoint)
 
+    def save_sharded_model(self, model: GeminiDDP, checkpoint_path: str, gather_dtensor: bool = False, variant: Optional[str] = None, max_shard_size: int = 1024, use_safetensors: bool = False):
+        """
+        Save sharded model
+        """
+        state_dict_shard = model.state_dict_shard(max_shard_size=max_shard_size, only_rank_0=False)
+        weights_name, save_index_file = get_base_filenames(variant, use_safetensors)
+        total_size = 0
+        single_shard = None
+        single_shard_file = None
+        index_file = CheckpointIndexFile(checkpoint_path)
+        for idx, shard_pair in enumerate(state_dict_shard):
+            if not self.coordinator.is_master():
+                continue
+            shard = shard_pair[0]
+            shard_file = get_shard_filename(weights_name, idx)
+            total_size = total_size + shard_pair[1]
+            for key in shard.keys():
+                index_file.append_weight_map(key, shard_file)
+            if idx == 0:
+                single_shard = shard
+                single_shard_file = get_shard_filename(weights_name, idx)
+                continue
+            if idx == 1:
+                checkpoint_file_path = os.path.join(checkpoint_path, single_shard_file)
+                save_state_dict(single_shard, checkpoint_file_path, use_safetensors)
+                single_shard = None
+                single_shard_file = None
+            
+            total_size = total_size + shard_pair[1]
+            checkpoint_file_path = os.path.join(checkpoint_path, shard_file)
+            save_state_dict(shard, checkpoint_file_path, use_safetensors)
+
+        if  single_shard is not None:
+            checkpoint_file_path = os.path.join(checkpoint_path, weights_name)
+            save_state_dict(single_shard, checkpoint_file_path, use_safetensors)
+            return
+        
+        index_file.append_meta_data("total_size", total_size)
+        index_file.write_index_file(save_index_file)
+        logging.info(
+            f"The model is going to be split to checkpoint shards. "
+            f"You can find where each parameters has been saved in the "
+            f"index located at {save_index_file}."
+        )
+
+
+    def load_sharded_model(self, model: nn.Module, checkpoint_index_file: Path, strict: bool = False, use_safetensors: bool = False):
+        return super().load_sharded_model(model, checkpoint_index_file, strict, use_safetensors)
```

Comment:
GeminiDDP overwrites `_load_from_state_dict()` in a non-recursive fashion. We can not simply reuse `super().load_sharded_model()`